libraries/bot-azure/src/main/java/com/microsoft/bot/azure/blobs/BlobsTranscriptStore.java

```diff
@@ -0,0 +1,423 @@
+        }
+    }
+
+    public CompletableFuture<PagedResult<Activity>> getTranscriptActivities(String channelId, String conversationId,
+                                                                            @Nullable String continuationToken,
+                                                                            OffsetDateTime startDate) {
+        if (startDate == null) {
+            startDate = OffsetDateTime.now(ZoneId.of("UTC"));
+        }
+
+        final int pageSize = 20;
+
+        if (StringUtils.isBlank(channelId)) {
+            throw new IllegalArgumentException("Missing channelId");
+        }
+
+        if (StringUtils.isBlank(conversationId)) {
+            throw new IllegalArgumentException("Missing conversationId");
+        }
+
+        PagedResult<Activity> pagedResult = new PagedResult<Activity>();
+
+        String token = null;
+        List<BlobItem> blobs = new ArrayList<BlobItem>();
+        do {
+            String prefix = String.format("%1$s/%2$s/", sanitizeKey(channelId), sanitizeKey(conversationId));
+            Iterable<PagedResponse<BlobItem>> resultSegment = containerClient.listBlobsByHierarchy(prefix)
+                .iterableByPage(token);
+            token = null;
+
+            for (PagedResponse<BlobItem> blobPage: resultSegment) {
+                for (BlobItem blobItem: blobPage.getValue()) {
+                    DateTime parseDateTime = DateTime.parse(blobItem.getMetadata().get("Timestamp"));
+                    Instant instantTime = Instant.ofEpochMilli(parseDateTime.getMillis());
+                    OffsetDateTime offsetParserDateTime = OffsetDateTime.ofInstant(instantTime, ZoneId.of("UTC"));
+                    if (offsetParserDateTime.isAfter(startDate) || offsetParserDateTime.isEqual(startDate)) {
+                        if (continuationToken != null) {
+                            if (StringUtils.equals(blobItem.getName(), continuationToken)) {
+                                // we found continuation token
+                                continuationToken = null;
+                            }
+                        } else {
+                            blobs.add(blobItem);
+                            if (blobs.size() == pageSize) {
+                                break;
+                            }
+                        }
+                    }
+                }
+
+                // Get the continuation token and loop until it is empty.
+                token = blobPage.getContinuationToken();
+            }
+        } while (!StringUtils.isBlank(token) && blobs.size() < pageSize);
+
+        pagedResult.setItems(blobs
+            .stream()
+            .map(bl -> {
+                BlobClient blobClient = containerClient.getBlobClient(bl.getName());
+                return this.getActivityFromBlobClient(blobClient);
+            })
+            .map(CompletableFuture::join).collect(Collectors.toList()));
```

Comment:
```suggestion
            .map(CompletableFuture::join).collect(Collectors.toList()));
```

Are you okay about this `.map` because [C#](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.Azure.Blobs/BlobsTranscriptStore.cs#L211) returns the result of them. I don't know if it's a property or just remove the nulls/empties values